app/src/test/java/com/douglasbruce/tasky/core/domain/validation/EmailValidatorTests.kt

```diff
@@ -0,0 +1,52 @@
+package com.douglasbruce.tasky.core.domain.validation
+
+import com.google.common.truth.Truth
+import org.junit.Before
+import org.junit.Test
+
+internal class EmailValidatorTests {
+    private lateinit var emailValidator: EmailValidator
+    private lateinit var emailMatcher: EmailMatcher
+
+    @Before
+    fun setUp() {
+        emailMatcher = object : EmailMatcher {
+            override fun matches(email: String): Boolean {
+                return email.contains("@") && email.contains(".")
+            }
+        }
+        emailValidator = EmailValidator(emailMatcher)
+    }
+
+    @Test
+    fun `Invalid email missing @ and dot, returns false with format error`() {
+        val email = "test"
+        val result = emailValidator(email)
+        Truth.assertThat(result.successful).isFalse()
+        Truth.assertThat(result.errorType).isEquivalentAccordingToCompareTo(ErrorType.FORMAT)
+    }
+
+    @Test
+    fun `Invalid email missing @, returns false with format error`() {
+        val email = "test.com"
+        val result = emailValidator(email)
+        Truth.assertThat(result.successful).isFalse()
+        Truth.assertThat(result.errorType).isEquivalentAccordingToCompareTo(ErrorType.FORMAT)
+    }
+
+    @Test
+    fun `Invalid email missing dot, returns false with format error`() {
+        val email = "test@tester"
+        val result = emailValidator(email)
+        Truth.assertThat(result.successful).isFalse()
+        Truth.assertThat(result.errorType).isEquivalentAccordingToCompareTo(ErrorType.FORMAT)
+    }
+
```

Comment:
This test class tests nothing, but your anonymous class created in the test (and that your validator actually calls the `matches` function). If your real `EmailMatcherImpl` would fail, this test class would not catch it, since it doesn't use such an instance.

The anonymous class instance here would be a fake in the end, but a fake only makes sense if it's used to isolate a class under test. In your case you're testing the logic of the fake though. For your email matcher, you don't need a test, since all it does is calling the email matcher function from the Android SDK - which you can be sure was thoroughly tested by Google.

Reply:
I was initially wasn't going to bother with this test class for this exact reason. I thought asserting the expected outcomes  would provide enough value to warrant the test class but maybe not. Will remove in a following PR